app/src/main/java/com/calendar/MainActivity.java

```diff
@@ -129,9 +132,22 @@ private void getEventsFromSnapshot(DataSnapshot dataSnapshot, String fromDate) {
         for (DataSnapshot snapshot : dataSnapshot.child(fromDate).getChildren()) {
             Event event = new Event(
```

Comment:
Woah, it's almost unable to understand... Mbe it's better to save this all statements in class constructor to variables first...

Reply:
Well it will still be a mess, but I think it is worth a try